Use find-babel-config for the babel config lookup

tleunen opened the pull request:
This will fix #293. The module `find-babel-config` searches for the closest babel configuration (`.babelrc` or babel in `package.json`) starting in a specified directory.

I've tested the code against the react-storybook-demo and everything looks fine. I checked that the path was correct and the check `inConfigDir` is also correct based on the found configuration.

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Add shortcuts to go back and forth (prev & next) (#176)

Add shortcuts to jump to previous and next stories

arunoda committed:
Update CHANGELOG

arunoda committed:
1.23.0

arunoda committed:
Add power tools link

arunoda committed:
Remove NODE_ENV=production flag.
Also added some checks for react-hmre existance and removed it.
It doesn't work well with React Storybook and we don't need it.

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Merge branch 'master' of github.com:kadirahq/paper-ui

arunoda committed:
Make left-panel scrollable.

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Fix double scrollbar issue in fullscreen mode. See #179.

arunoda committed:
Update CHANGELOG.md

arunoda committed:
1.24.0

arunoda committed:
Update Changelog for the version.

witnessmenow committed:
Fix for #183 - issue with removing a preset from users babelrc

stefvhuynh committed:
Fix babel extends issue by replacing with absolute paths

arunoda committed:
Make qs a dependency.

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Fix issue related to previous commit.

jamieholliday committed:
Make left panel stories scrollable (#182)

arunoda committed:
Update CHANGELOG.

arunoda committed:
Merge with master.

arunoda committed:
Merge branch 'sevenleaps-master'

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Merge with master.

arunoda committed:
Merge branch 'stefvhuynh-master'

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Update scripts to make them compatible with linux.

arunoda committed:
Update CHANGELOG.md

arunoda committed:
1.25.0

arunoda committed:
Update CHANGELOG.md

Commit:
Add npm version badge to README (#193)

This project is moving quickly, and it would be useful for the README to indicate the currently-available version on npm. This branch adds the common `https://badge.fury.io/` image, similar to other React projects on GitHub.

thebuilder committed:
Ensure asset directory exists (#194)

* Ensure asset directory exists

Creating a static build would fail (Error: ENOENT: no such file or directory) if it encountered an asset in a subdirectory.
Use mkdir to ensure the path exists, before writing the file.

For example, referencing fonts would fail:
```
@font-face {
  font-family: 'vaud';
  src: url('../fonts/vauddisplaybold-webfont.eot');
  src:
    url('../fonts/vauddisplaybold-webfont.eot?#iefix') format('embedded-opentype'),
    url('../fonts/vauddisplaybold-webfont.woff') format('woff'),
    url('../fonts/vauddisplaybold-webfont.ttf') format('truetype'),
    url('../fonts/vauddisplaybold-webfont.svg#vaud_displayregular') format('svg');
  font-weight: 700;
  font-style: normal;
}
```

* Make ESlint happy

arunoda committed:
Update CHANGELOG.md

arunoda committed:
1.26.0

arunoda committed:
Add build files.

Commit:
feat(host): ability to control host name

Commit:
fix(typo)

Commit:
Merge pull request #195 from zetxx/master

feat(host): ability to control host name

thani-sh committed:
Listen on all hosts by default

Commit:
Merge pull request #198 from kadirahq/listen-on-all

Listen on all hosts by default

arunoda committed:
Add stack-source-map again but only for Chrome.

arunoda committed:
Merge branch 'better-errors'

arunoda committed:
Update Changelog.

arunoda committed:
Fix lint errors.

arunoda committed:
Update about storybooks-core

arunoda committed:
Update how_storybook_works.md

Commit:
Move modules to storybook-core repo (#196)

* Move modules to storybook-core repo

* Use keyEvents from @kadira/storybook-core

arunoda committed:
Update CHANGELOG

arunoda committed:
1.27.0

thani-sh committed:
Export storybook middleware

Commit:
Merge pull request #211 from kadirahq/storybook-mw

Export storybook middleware

thani-sh committed:
Build source files

thani-sh committed:
Update CHANGELOG

thani-sh committed:
Release v1.28.0

thani-sh committed:
Add charset meta tag

Commit:
Merge pull request #216 from kadirahq/add-charset

Add charset meta tag

thani-sh committed:
Build source files

thani-sh committed:
Update CHANGELOG

thani-sh committed:
Release v1.28.1

izzi-bikun-kr committed:
Make bundle location relative to webpack publicPath

TigerC10 committed:
Switching to stage-0 support

Commit:
Merge pull request #226 from bmagnus/public-path-pr

Make bundle location relative to webpack publicPath

thani-sh committed:
Build source files

thani-sh committed:
Release v1.28.2.

thani-sh committed:
Join path using the path module

thani-sh committed:
Use relative public path

Resolve from html files path

thani-sh committed:
Revert PR #226

thani-sh committed:
Update Changelog

thani-sh committed:
Release v1.28.2.

thani-sh committed:
Release v1.28.3.

thani-sh committed:
Copy manager.js only once

thani-sh committed:
Use public path from webpack config

thani-sh committed:
Build source files

thani-sh committed:
Merge branch 'fix-custom-public-path'

thani-sh committed:
Update CHANGELOG

thani-sh committed:
Release v1.28.4.

thani-sh committed:
Fix lint errors

Commit:
Merge pull request #228 from TigerC10/master

Switching to stage-0 support

thani-sh committed:
Update CHANGELOG

thani-sh committed:
Release v1.28.5.

thani-sh committed:
Support multiple static dirs

Commit:
Merge pull request #233 from kadirahq/multiple-static-dirs

Support multiple static dirs

thani-sh committed:
Update CHANGELOG

thani-sh committed:
Release v1.29.0.

thani-sh committed:
Copy missing manager.js.map file

The sourcemap file for manager.js was not copied when
building the static storybook bundle. Which seems to cause
issues with Safari browser.

The static build seems to work fine on Safari 9. But better
fix the issue anyways.

Fixes #234

Commit:
Merge pull request #236 from kadirahq/cp-manager-srcmap

Copy missing manager.js.map file

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.29.1

thani-sh committed:
Add es6 shim

thani-sh committed:
Fix path on Windows

Commit:
Merge pull request #240 from kadirahq/fix-path-join

Fix path on Windows

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.29.2

TigerC10 committed:
Updating dependency on storybook-core

Commit:
Merge pull request #241 from TigerC10/master

Updating dependency on storybook-core

Commit:
Merge pull request #238 from kadirahq/add-es6-shim

Add es6 shim by default

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.29.3

thani-sh committed:
Copy html files after static files

Fixes #242

mnmtanish committed:
Merge pull request #243 from kadirahq/protect-html-files

Copy html files after static files

thani-sh committed:
Build Source files

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.29.4

thani-sh committed:
Build source files

thani-sh committed:
Update dependency versions

mnmtanish committed:
Merge pull request #246 from kadirahq/fix-filter-crash

Fix filter crash with '\' character

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.29.5

thani-sh committed:
Fuzzy filter kinds

mnmtanish committed:
Merge pull request #247 from kadirahq/fuzzy-filter

Fuzzy filter kinds

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.30.0

arunoda committed:
Update configure_storybook.md

thani-sh committed:
Pass selected kind and story as args

mnmtanish committed:
Merge pull request #250 from kadirahq/context-argument

Pass selected kind and story as args

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.31.0

arunoda committed:
Simplyfy the manager UI.

arunoda committed:
use the manager UI directly from the storybook-ui

arunoda committed:
Write completely in ES2015.

arunoda committed:
Use @kadira/storybook-ui

arunoda committed:
Add dist.

arunoda committed:
1.32.0

arunoda committed:
Update CHANGELOG.

arunoda committed:
Extend the Storybook UI provider from the base Provider.

arunoda committed:
1.32.1

arunoda committed:
Update CHANGELOG

arunoda committed:
Implement the new addon API.

arunoda committed:
Tested addon and decorator API with a real app.

arunoda committed:
Add context support to decorators.

arunoda committed:
Add kind to the api.

arunoda committed:
Create ROADMAP.md

arunoda committed:
Merge branch 'addons'

arunoda committed:
Implement a extension API (#258)

* Implement the new addon API.

* Tested addon and decorator API with a real app.

* Add context support to decorators.

* Add kind to the api.

* Add extension documentation.

* Update extension docs.

* Fix a small typo.

arunoda committed:
Merge branch 'master' of github.com:kadirahq/react-storybook

arunoda committed:
1.33.0

arunoda committed:
Update CHANGELOG

arunoda committed:
Update storybook-ui to v2.2.0 (Which has the shortcut permalinks).

arunoda committed:
1.34.0

arunoda committed:
Update CHANGELOG.md

jurgob committed:
Update extensions.md (#262)

pcardune committed:
Don't always override NODE_ENV in build-storybook (#272)

My webpack configuration uses the `DefinePlugin` to make `process.env.NODE_ENV` available to my react components, which in turn optionally do different things depending on the NODE_ENV (related to debugging and such).

Unfortunately, these development-only features don't show up when I use `build-storybook` because `build-storybook` overrides `process.env.NODE_ENV`.

While I think it is quite reasonable for `build-storybook` to default to a "production ready" build, it should not unconditionally override the `NODE_ENV` environment variable to do so.

This change makes `"production"` the default, but also allows people to set a different `NODE_ENV` value and have it get used.

arunoda committed:
1.34.1

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Update dist.

arunoda committed:
Update storybook-ui to v2.3.0

arunoda committed:
1.35.0

arunoda committed:
Update Changelog.md

arunoda committed:
Fix issue #275 of bad error handling.

arunoda committed:
1.35.1

arunoda committed:
Update Changelog.

Maxhodges committed:
remove semicolon

fixed typo

mnmtanish committed:
Merge pull request #282 from Maxhodges/patch-1

remove semicolon

arunoda committed:
Update configure_storybook.md

arunoda committed:
Add fontFamily to the error display.

arunoda committed:
Merge branch 'master' of github.com:kadirahq/react-storybook

arunoda committed:
1.35.2

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Fix the lint error.

Tarabyte committed:
React intl addon (#288)

* Fix typo "togheter" --> "together"

* Adds link to react-storybook-addon-intl

lukesneeringer committed:
Support watchOptions configuration. (#287)

This commit causes the watchOptions in the configuration provided
to `react-storybook` to be passed on to `webpack-dev-middleware`.

This allows those who need these options for hot reloading to be able
to use them again.

arunoda committed:
1.36.0

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Add dist.

arunoda committed:
Update storybook-ui to 2.4.0

arunoda committed:
1.37.0

arunoda committed:
Update CHANGELOG.md

nadav-dav committed:
Added npm-shrinkwrap.json to npmignore (#305)

arunoda committed:
Deleted npm-shrinkwrap

arunoda committed:
Merge branch 'master' of github.com:kadirahq/react-storybook

arunoda committed:
1.38.0

arunoda committed:
Update CHANGELOG.md

roonyh committed:
Wrote prepublish in javascript

This allows storybook to be built in windows too.

mnmtanish committed:
Merge pull request #308 from kadirahq/prepublish-in-js

Wrote prepublish in javascript

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.38.1

thani-sh committed:
Update dist

thani-sh committed:
Fix prepublish script

mnmtanish committed:
Merge pull request #319 from kadirahq/fix-prepublish

Fix prepublish script

thani-sh committed:
Update dist

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.38.2

thani-sh committed:
Add names for action and linkTo functions

mnmtanish committed:
Merge pull request #321 from kadirahq/add-function-names

Add names for action and linkTo functions

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.38.3

arunoda committed:
Update @kadira/storybook-ui to v2.6.0

arunoda committed:
Update CHANGELOG.md

arunoda committed:
1.39.0

orta committed:
Add react-storybooks-relay-container (#329)

Commit:
Use find-babel-config for the babel config lookup

find-babel-config searches for the configuration in the closest .babelrc or
package.json file

arunoda commented on src/server/config.js:
```diff
@@ -51,17 +36,14 @@ function loadBabelConfig(babelConfigPath) {
-  let inConfigDir = true;
-
-  if (!babelConfig) {
-    babelConfig = loadBabelConfig('.babelrc');
```
Here we are trying to load two babelConfig files? First try that in the storybook config dir and then in the project root.

I think that logic is not done in this new setup.

> We've no idea where the configDir is located. By default it's `.storybook` but usually it's getting changed.

tleunen replied:
Oh gotcha. Can `.storybook` be outside of the project? What would be the reason for it?

Because `find-babel-config` will make a search recursively, so if it's still inside the project, everything should work as expected.
If not, I can change the PR to match what was done before.

arunoda replied:
@tleunen it could be. I don't know the reason some people just do it.
May be you need to do another lookup of `find-babel-config` with depth=0 for the config directory.

tleunen replied:
Yep, for a safer solution, I'll change the code to keep the 2 calls and set them with a `depth=0` ;)

thani-sh committed:
Sort devDependencies

thani-sh committed:
Update babel-* modules

thani-sh committed:
Update dist

thani-sh committed:
Remove info HMR logs

thani-sh committed:
Add getStoryBook function

mnmtanish committed:
Merge pull request #325 from kadirahq/update-babel

Update babel-* modules

mnmtanish committed:
Merge pull request #331 from kadirahq/quieter-hmr

Remove info HMR logs

thani-sh committed:
Update dist

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.39.1

thani-sh committed:
Clear decorators on module dispose

The `config.js` file acts as the main entrypoint for users storybook
code. Stories, decorators etc. are added from here. When webpack HMR
replaces this module global decorators get applied repeatedly. This
happens because although the config.js file gets replaced the storybook
module remains unchanged along with all global decorators we may have
added before the replacement.

It is okay to remove all decorators when config.js module gets replaced
because we can be sure that they'll be added again when config.js file
gets executed.

mnmtanish committed:
Merge pull request #335 from kadirahq/fix-hmr-duplicate-decorator

Clear decorators on module dispose

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.40.0

thani-sh committed:
Support requireing from node

Wrap calls to browser only features to allow storybook module
to be required from node. This can be used to build CLI tools.

thani-sh committed:
Explain why pageBus can be null

mnmtanish committed:
Merge pull request #337 from kadirahq/fix-node-require

Support requireing from node

thani-sh committed:
Add unit tests

thani-sh committed:
Rename getStoryBook to getStorybook

mnmtanish committed:
Merge pull request #332 from kadirahq/get-storybook

API to get storybook stories

thani-sh committed:
Update CHANGELOG

thani-sh committed:
1.41.0

thani-sh committed:
Build dist with newer React

arunoda committed:
Update defaults to match create-react-app (#342)

* Match default setup same as create-react-app.

* Stop pre-building the manager
This allow us to customize the manager dynamically.

* Update manager for the production build.

* Remove building source-maps.

* Fix lint issues.

* Improve static file handling.

* Update dev docs.

* Use dist directory for the manager's source.

* Add support for stage-0.

* Replace raw loader with the css loader in production CSS handling.

* Remove specific babel plugins since we support stage-0.

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Update @kadira/storybook-ui

arunoda committed:
2.0.0

arunoda committed:
Update Changelog.md

arunoda committed:
Update build.js

arunoda committed:
Update docs to reflect recent changes.

arunoda committed:
Update docs to reflect recent changes.

mnmtanish committed:
[WIP] Use the new addons API (#346)

* Update Storybook UI

* Setup channel for preview

* Load addons from user or load default addons

* Allow easy importing default set of addons

For an example addons.js:

```js
import '@kadira/storybook/addons';
import '@kadira/storybook-addon-hello/register';

```

* Export correct action, linkTo functions

* Rename pageBus to channel

* Fix tests and lint errors

* Update some comments.

arunoda committed:
2.1.0

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Remove default webpack config for all config types. (#348)

Now we only use the Create React App based config
if there's no custom webpack config.
This will fix issues like #347.

arunoda committed:
2.1.1

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Fix lint issues.

arunoda committed:
Add node_modules as a excludePath for js. (#353)

arunoda committed:
Use the default babel-config from CRA. (#354)

* Use the default babel-config from CRA.

* Fix lint issues

arunoda committed:
Use es6-shim directly into webpack config. (#355)

arunoda committed:
Add NPM2 support. (#356)

* Add NPM2 support.
With the use alot of require.resolve and
defining the missing redux dependency.

* Fix lint issues.

* Update storybook-ui which is optimized for NPM2

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Add autofixer defaults. (#357)

See: https://github.com/facebookincubator/create-react-app/pull/345

arunoda committed:
Update CHANGELOG.md

arunoda committed:
2.2.0

mnmtanish committed:
Use @kadira/storybook-channel modules (#359)

* Use @kadira/storybook-channel module

* Update channel addons

arunoda committed:
Update to use the latest storybook-ui.

arunoda committed:
2.2.1

arunoda committed:
Update CHANGELOG.md

arunoda committed:
Add logo.

mattkime committed:
remove extract-text-webpack-plugin dep (#369)

arunoda committed:
2.2.2

arunoda committed:
Update CHANGELOG.md

lukyth committed:
Add new item to extension list

Add mthuret/storybook-addon-specifications to extension list in docs/extensions.md

arunoda committed:
Update CHANGELOG.md

mnmtanish committed:
Merge pull request #377 from lukyth/patch/extension

Add new item to extension list

arunoda committed:
Update CHANGELOG.md
